DNA_RNA/DNA_RNA_HW.py

```diff
@@ -0,0 +1,83 @@
+#Создать классы Dna и Rna, объекты которых обладают следующими свойствами:
+
+#    Могут быть инициализированы строкой, соответствующей последовательности ДНК/РНК
+#    Позволяют определить долю GC нуклеотдиов в последовательности с помощью метода gc_content
+#    Позволяют создать комплементарную последовательность с помощью метода reverse_complement
+#    Позволяют итерироваться по последовательности
+#    Объекты с одинаковыми последовательностями должны быть равны (так как про eq мы поговорить не успели еще, по собственному желанию)
+#    Их можно добавлять в множества (так как про hash мы поговорить не успели еще, по собственному желанию)
+
+
+#Объекты класса Dna дополнительно имеют метод transcribe позволяющий создавать объект класса Rna, соответствующий результату транскрипции последовательности.
+
+#Весь код написать в отдельной ветке в своем репозитории и сделать PR в main, поставив меня/Катю в ревьюеры.
+
+
+
+class NucleicAcid:
+    def __init__(self, sequence):
+        self.sequence = sequence
+
+    def gc_content(self):
+        seq = list(self.sequence.upper())
+
+        g_nucleotides = seq.count('G')
+        c_nucleotides = seq.count('C')
+
+        total_nucleotides_number = len(self.sequence)
+
+        gc_content =  ((g_nucleotides + c_nucleotides) / total_nucleotides_number) * 100
+
+        return gc_content
+
+    def __iter__(self):
+        return NucleicAcidIterator(self)
```

Comment:
можно было и чуть проще
```suggestion
    def __iter__(self):
        return iter(self.sequence)
```